iDiff/utils/package_diff_utils.go

```diff
@@ -1,11 +1,31 @@
 package utils
 
 import (
+	"fmt"
 	"io/ioutil"
 	"path/filepath"
 	"reflect"
+	"strings"
+
+	"github.com/golang/glog"
 )
 
+// MultiVersionPackageDiff stores the difference information between two images which could have multi-version packages.
+type MultiVersionPackageDiff struct {
+	Image1    string
+	Packages1 map[string]map[string]PackageInfo
+	Image2    string
+	Packages2 map[string]map[string]PackageInfo
+	InfoDiff  []MultiVersionInfo
+}
+
+// MultiVersionInfo stores the information for one multi-version package in two different images.
```

Comment:
The multi-version additions are intended to handle the case of multiple versions of the same package within one image, which then need to be compared with another image's packages which may also have multiple versions.